modules/teacher_emails/email_finder.py

```diff
@@ -0,0 +1,90 @@
+import os
+from modules.teacher_emails.teacher_weights import TeacherWeights
+from typing import Iterable, Set, List, Optional
+from modules.teacher_emails.professor import Professor
+import discord
+import psycopg2.extensions as sql
+
+
+class EmailFinder:
+	def __init__(self, conn: sql.connection) -> None:
+		self.conn = conn
+		self.search_weights = {
+			"keyword": 1,
+			"curr_channel": 2,
+			"mentioned_channel": 10,
+			"mentioned_teacher": 20,
+		}
+
+	def search(
+		self,
+		query: List[str],
+		mentioned_channels: List[discord.TextChannel],
+		curr_channel: discord.TextChannel,
+		mentioned_members: List[discord.Member],
+	) -> Set[Professor]:
+		"""returns a list of professors who best match the query"""
+		teachers = TeacherWeights()
+
+		# add all teachers found by mention of the channel of a course they teach
+		for channel in mentioned_channels:
+			for t_id in self.__search_channel(channel.id):
+				teachers[t_id] += self.search_weights["mentioned_channel"]
+
+		# add the teachers of the current channel
+		for t_id in self.__search_channel(curr_channel.id) or []:
+			teachers[t_id] += self.search_weights["curr_channel"]
+
+		# add the teacher found (if any) by mention of their discord accont
```

Comment:
typo in comment

Reply:
for some reason my spell checker thought that was a real word... its not in my list of added words... weird...
++search urban dictionary accont